### ISSUE TITLE
Validate the artifact name in `gto.api.register()/promore()/annotate()`

### ISSUE THREAD

**Issue by Suor** [iterative/gto#172]
Currently I get a lower level `git.exc.GitCommandError` with `fatal: 'model a@v1.2.3' is not a valid tag name` message.

**Comment by aguschin** [comment]
@Suor, Can't reproduce this. If I try
```
>>> gto.api.promote(".", "m ", promote_ref="HEAD", stage="prod")
```
I get
```
ValidationError: Invalid value 'm '. Only alphanumeric characters, '-', '/' are allowed.
```
Same for `gto.api.register`.

Are you using the latest GTO version? Could you please post some code to reproduce?

**Comment by Suor** [comment]
Ok, I was using `gto` version `0.1.9`. It looks fixed in `0.1.10`.
